File: backend/src/main/java/com/apocalypse/thefall/mapper/TerrainConfigurationMapper.java
package com.apocalypse.thefall.mapper;

import com.apocalypse.thefall.dto.TerrainConfigurationDto;
import com.apocalypse.thefall.model.TerrainConfiguration;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface TerrainConfigurationMapper {

    TerrainConfigurationDto toDto(TerrainConfiguration terrain);

    List<TerrainConfigurationDto> toDtoList(List<TerrainConfiguration> terrains);
}

File: backend/src/main/java/com/apocalypse/thefall/mapper/TerrainConfigurationMapper.kt
package com.apocalypse.thefall.mapper

import com.apocalypse.thefall.dto.TerrainConfigurationDto
import com.apocalypse.thefall.entity.TerrainConfiguration

fun TerrainConfiguration.toDto(): TerrainConfigurationDto =
    TerrainConfigurationDto(
        name = this.name,
        descriptions = this.descriptions,
        movementCost = this.movementCost,
        walkable = this.walkable,
        path = this.path
    )

fun List<TerrainConfiguration>.toDtoList(): List<TerrainConfigurationDto> = this.map { it.toDto() }
